raft-tech/TANF-app
# feat(1358): create default ACF OCIO permissions group

## Summary of Changes

This PR adds a new default "ACF OCIO" permissions group that is restricted to read-only access for the ZAP and ClamAV objects in Django Admin. Users with this role will also be redirected to Django Admin upon subsequent login attemts from the frontend, while retaining access to the frontend Home and Profile pages.

Pull request closes #1358 

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d 
```

1. Open http://localhost:3000/ and sign in.
1. Proceed with functional tests as described herein.
1. Test steps should be captured in the demo GIF(s) and/or screenshots below.
	1. Login to the app as a new user
	2. Go to Django Admin and grant this user the ACF ACIO group
	3. Ensure the user has read-only access to the ClamAV and ZAP tables in Django Admin
	4. Ensure the user can only access the Home and Profile pages in the frontend
> *Demo GIF(s) and screenshots for testing procedure*

![Screen Shot 2022-03-03 at 2 45 28 PM](https://user-images.githubusercontent.com/12901172/156641135-0aff7984-eb53-4c25-9f68-1e29e64e3e44.png)
![Screen Shot 2022-03-03 at 2 46 06 PM](https://user-images.githubusercontent.com/12901172/156641240-99756863-1176-4aaa-8127-48e93f1f73e4.png)
![Screen Shot 2022-03-03 at 2 47 14 PM](https://user-images.githubusercontent.com/12901172/156641416-ceadf8e1-c4e7-4f21-ab95-18a1850af2d7.png)

## Tabletop


## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] New ACF OCIO group is created in our database by default
- [x] New group does not have STTs nor Regions associated to it
- [x] ACF OCIO group should have read-only access to DAC security [scans](https://tdp-backend-staging.app.cloud.gov/admin/security/) (e.g. OWASP scans, ClamAV scans)
- [x] ACF OCIO group should not have access to view other areas of DAC nor the Frontend where feasible

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_93.39%_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_97.07%%_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

_n/a_


### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

_no new issues_
_no new issues_
### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

_n/a_
